Mathematics/TriangleArea/triangle-area.ipynb

```diff
@@ -0,0 +1,1150 @@
KP4eE8FOrbGN98M+LZZ/MAQM8TygCgAan4Y/PmiIGBcr5W8cehQ3kAoKcJZQDQkLlzyxWzSnvxxwsvRHzjG3kAoGcJZQDQoLvuili6NA+FFMyq82XJU09FnD+fBwB6klAGAA1LxR/z5+eh8M47V4PZxYsRjz2m+AOglwllANCw9xXfjbduLc+ZVerbGN96S/EHQC8TygCgAwwNlcFscLCcFX8A9A+hDAA6RCr+WLUqD4X6jaUTxR8AvUkoA4AOsmxZ+aik1TLFHwC9TSgDgA6TVstuvz0PhVT8UVH8AdB7hDIA6DDpXNm2bSOLP+rBTPEHQG8RygCgA6VAloKZ4g+A3ieUAUCHSlsYN24svlnn79aKPwB6k1AGAB3szjsjli/PQyGtltWDmeIPgO4nlAFAh3vggYhFi/JQqDcyKv4A6H5CGQB0gQ0bIoaH81Cony9LxR979oxcQQOgewhlANAFZswoiz8GBsq5vfjjlVcivva1PADQVYQyAOgSt90WsXZtHgrtxR8vvqj4A6AbCWUA0EWWLo245548FOrny5KdOyPefDMPAHQFoQwAusyqVRHz5+ehkG4sXQWzy5cjdu1S/AHQTYQyAOhCmzcr/gDoFUIZAHShVPyxZUvxjTx/J1f8AdC9hDIA6FJz5kQ8+GAeCoo/ALqTUAYAXWzJkojFi/NQUPwB0H2EMgDocuvXjyz+qG9jVPwB0PmEMgDoculc2datEbNmlXNaKUuNjBXFHwCdTSgDgB4wNBSxbVtZAJIo/gDoHkIZAPSI2bPLrYwVxR8A3UEoA4AecscdEfffn4dCWi2rBzPFHwCdRygDgB6TQtmCBXkoKP4A6GxCGQD0oE9+MmJ4OA8FxR8AnUsoA4AeNDgYsWVLxMBAOSv+AOhcQhkA9Kg5cyIefjgPhWsVf5w6lQcAGiOUAUAPu/POiJUr81BIq2Vp1azy+OMR587lAYBGCGUA0ONWrChbGSv182Wp+OPJJ0deA2B6CWUA0AfS/ctmzsxDoR7CTp+O2Ls3DwBMO6EMAPrAjBkRmzcX3/jzd/724o8TJyKOHMkDANNKKAOAPjFvXsSDD+ah0F788fzzij8AmiCUAUAfWbIk4p578lBoL/7YvTvi4sU8ADAthDIA6DOrVkUsXJiHQjpfVgWzCxfKYKb4A2D6CGUA0IfSjaVnzcpDoX6+7OTJiP378wDAlBPKAKAPDQ5GPPJI8UIgvxJoL/44erR8ADD1hDIA6FNz5kQ88EAeCtcq/kirZgBMLaEMAPrY8uURixfnoVAv/kgBbceO8pwZAFNHKAOAPpduLJ3q8iv14o/LlyOeekrxB8BUEsoAoM+lc2XpxtLDw/lCQfEHwPQRygCAmDmzDGYDA+Ws+ANg+ghlAEDL3LkRa9fmoaD4A2B6CGUAwLuWLo1YtiwPhfbij3Rj6XPnyhmAySGUAQAjrFkTsXBhHgr14o+LFyN27Rq5ggbAxAhlAMAIqfhjy5aI2bPzhUL9fNmZM4o/ACaTUAYAvMfgYBnMhobKub3448iRiOPH8wDAhAhlAMA1pZWyhx8uV86S9uKPvXsjTp/OAwA3TSgDAK5r0aKIFSvyUEirZVUwS887dpTnzAC4eUIZAHBDKZTNn5+HQn0bYwpk27dHXLmSLwAwbkIZADCqTZvKG0xXUiNj5ezZiIMH8wDAuAllAMCoUiDbvDliYKCc24s/Dh+OeO21PAAwLkIZADAm8+ZFrF6dh0J78cfOnYo/AG6GUAYAjNmyZRFLl+ahkFbLqhtLK/4AuDlCGQAwLuvWRSxcmIdC/XyZ4g+A8RPKAIBxSfctSzeWTvcxq7QXfzzzTB4AGJVQBgCM2+BgGcyGhsq5vfjjxImIV1/NAwA3JJQBADclrZRt2FCunCXpTFk9mO3eHXHmTB4AuC6hDAC4aXfcEXHvvXkopGCm+ANgfIQyAGBCVq6MWLQoD4V0vqwKZhculFX59ep8AEYSygCACUs3lq4Xf9S3MZ46FbFvXx4AeA+hDACYsNGKP44di3jllTwAMIJQBgBMimsVf9S3LT79dMTJk3kA4F1CGQAwaVLxx/LleSik1bIqmKWzZqmR8dKlcgagJJQBAJMqFX/MnZuHQgpmVfFHamJ89NGIK1fKGQChDACYZGn74tatETNn5guF+vmys2cj9u/PAwBCGQAw+VIgS42MAwPl3F788fLLEceP5wGgzwllAMCUmDcvYvXqPBTaiz/27o04fToPAH1MKAMApsyyZeWjUj9flgLajh3lOTOAfiaUAQBTat26spWxkloY68UfKZilawD9SigDAKbcpk3lfcwq9fNlaQvjvn15AOhDQhkAMOUGByO2bIkYGirntFJWXx1LpR9HjuQBoM8IZQDAtEgrZQ8/XLz4yK8+UjCrF3+kmvxTp/IA0EeEMgBg2ixaFHHffXkopG2MVTBLz7t3R5w/X84A/UIoAwCmVQplqS6/Uj9fduGC4g+g/whlAMC0StsX042lh4fzhUI9hJ09G3HgQB4A+oBQBgBMu5kzy2A2MFDO6XxZfcUslX4cOpQHgB4nlAEAjZg7N2Lt2jwU0pmyevHHCy8o/gD6g1AGADRm6dKIZcvyUEirZdWNpZOdOxV/AL1PKAMAGrVmzcjij/r5sosXFX8AvU8oAwAalYo/tm6NmDUrXyi0F38891weAHqQUAYANG7GjIiNG8uAlrQXfxw7FnH0aB4AeoxQBgB0hLSF8aGHrgaz9uKP55+POHkyDwA9RCgDADrG3XdHrFiRh0JaLauCWXrevTvi0qVyBugVQhkA0FFSKFuwIA+FeiNjKv549NGIK1fKGaAXCGUAQMdJN5auF3/Uz5el4o/9+/MA0AOEMgCg4wwNRWzbFjE4WM5ppazeyPjyy+UDoBcIZQBAR5o9O2Lt2jwUUjCrF3/s2xfxxht5AOhiQhkA0LGWLIlYuTIPhfr5smTPnvKcGUA3E8oAgI6Wij9SK2MlbWOsF39s3674A+huQhkA0P
```

Comment:
This should be m^2, right?

I wonder if factoring out the common code between this and the other "what's the area" questions would:

1) Mean you have to write less code
2) Mean it'd be harder for this sort of a typo to happen (since the same thing would show up everywhere!).

 I'd be happy to help look into how this might work.

Reply:
Although I don't know where you found the typo, I agree that factoring out common code would be helpful in reducing the likeliness of typos. As these notebooks are already working, and now reviewed, I'm not sure if you want me to continue to work on these notebooks to update the code, or just reuse code when I can in future notebooks. I appreciate your feedback.

Reply:
I'm not sure how github associated my comment with that line :)

It was supposed to be associated with this line, which is asking for an area:

`if ans3.value == '7.52m' or ans3.value == '7.52 m':`

As for whether that kind of refactoring should happen before this goes in or after, I'd defer to @arcotbha 's thoughts here.

Reply:
(but the unit error, which rejects the correct answer and accepts an incorrect answer, should probably be fixed first  anyways :) )

Reply:
Probably the best way to approach this would be to more formally package the geogabra python wrapper, such that it would get installed into somewhere anything could import it.

@n-vachon : For context: Is this wrapper something you wrote yourself, or did it come from somewhere else?

Reply:
The GeoGebra python library was developed by one of our Callysto developers, and it's undergone several tweaks and revisions in the nbplus repository on Callysto. I think current best practice is to do a pip install.

Reply:
(ack, sorry, i've posted this geogabra-related comment on the wrong issue, so @sean-fitzpatrick 's reply is also in the wrong place).

Reply:
That unit error is still there, and whether it's resolved via a broader refactoring or just in the one-off case, it still needs to be resolved.